# Descarga De Datos

In [ ]:
from wget import download
import pandas as pd

if not path.exists("results.zip"):
    download("https://media.githubusercontent.com/media/StackExchange/Survey/6ae2209cd35a55e92739ba82f7887fb86807a069/packages/archive/2025/results.csv")
else:
    print("Archivo Ya Descargado")


if not path.exists("schema.csv"):
    download("https://media.githubusercontent.com/media/StackExchange/Survey/refs/heads/main/packages/archive/2025/schema.csv")
else:
    print("Archivo Ya Descargado")

df_results = pd.read_csv("results.zip", low_memory=False)
df_schema = pd.read_csv("schema.csv", low_memory=False)

# schema.csv: 
**Es el diccionario de datos de la encuesta. results.csv tiene las respuestas de cada persona, pero con nombres de columna abreviados.El schema te dice qué pregunta exacta corresponde a cada una de esas columnas.**

# results.csv:
**Contiene las respuestas individuales de la "Stack Overflow Developer Survey 2025", una encuesta anual que Stack Overflow realiza a la comunidad de desarrolladores sobre su perfil, trabajo, tecnologías y uso de herramientas de IA.**

# Visión general

**Dimensiones**

In [11]:
filas, columnas = df_results.shape
print(f"El Dataset \"results\" tiene {filas:,} filas y {columnas} columnas")
print(f"Memoria: {df_results.memory_usage(deep=True).sum() / 1e6:.1f} MB")
#df_results.head()

El Dataset "results" tiene 49,191 filas y 172 columnas
Memoria: 375.2 MB


**Tipos De Datos**

In [ ]:
df_results.info(verbose=True, show_counts=True)
print("\n")
tipos = (df_results.dtypes.astype(str).value_counts().rename_axis("Tipo de dato").reset_index(name="Cantidad de columnas"))
tipos


**Identificador, Duplicados Y Columnas Sin Valor**

In [ ]:
print("'ResponseId' es único?:", df_results["ResponseId"].is_unique)
print("Filas duplicadas:", df_results.duplicated().sum())
print("Columnas con un solo valor distinto:",[c for c in df_results.columns if df_results[c].nunique(dropna=True) <= 1])

# Interpretación

El dataset `results` contiene 49.191 respuestas (filas) y 172 variables (columnas), con un uso de memoria de aproximadamente 375 MB.

- **Tipos de datos:** 119 columnas son de texto (`str`), 52 son numéricas de punto flotante (`float64`) y 1 es entera (`int64`, el identificador `ResponseId`). Las numéricas son `float` y no `int` porque contienen valores nulos.
- **Identificador:** `ResponseId` es único, por lo que cada fila representa una respuesta distinta.
- **Duplicados:** no hay filas duplicadas.
- **Columnas sin información:** no hay columnas constantes ni vacías.
